OSDOCS-9882-bare-metal-hardware: Documented the 4.16 BM hardware prov bug fixes (#77714)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

[id="ocp-4-16-bare-metal-hardware-bug-fixes_{context}"]
==== Bare Metal Hardware Provisioning

* Previously, newer versions of Redfish used Manager resources to deprecate the Uniform Resource Identifier (URI) for the RedFish Virtual Media API. This caused any hardware that used the newer Redfish URI for Virtual Media to not be provisioned. With this release, the Ironic API identifies the correct Redfish URI to deploy for the RedFish Virtual Media API so that hardware relying on either the deprectaed or newer URI could be provisioned. (link: https://issues.redhat.com/browse/OCPBUGS-30171[*OCPBUGS-30171*]

> * Previously, the Bare Metal Operator (BMO) was not using a leader lock to control incoming and outgoing operator pod traffic. After a deployment included a new operator pod, the new pod competed with system resources, such as the `ClusterOperator` status, and this terminated any outgoing operator pods. This issue also impacted a cluster that did not include any bare-metal node. With this release, the BMO includes a leader lock to manage new pod traffic, and this fix resolves the competing pod issue. (link:https://issues.redhat.com/browse/OCPBUGS-25766[*OCPBUGS-25766*])

Reviewer: ```suggestion
* Previously, the Bare Metal Operator (BMO) was not using a leader lock to control incoming and outgoing operator pod traffic. After a deployment included a new operator pod, the new pod competed with system resources, such as the `ClusterOperator` status, and this terminated any outgoing operator pods. This issue also impacted clusters that do not include any bare-metal nodes. With this release, the BMO includes a leader lock to manage new pod traffic, and this fix resolves the competing pod issue. (link:https://issues.redhat.com/browse/OCPBUGS-25766[*OCPBUGS-25766*])
```

Reviewer: What is _deployment_ here? A `deployment` CR or an OpenShift deployment??
> After a deployment included a new operator pod

[id="ocp-4-16-bare-metal-hardware-bug-fixes_{context}"]
==== Bare Metal Hardware Provisioning

> * Previously, newer versions of Redfish used Manager resources to deprecate the Uniform Resource Identifier (URI) for the RedFish Virtual Media API. This caused any hardware that used the newer Redfish URI for Virtual Media to not be provisioned. With this release, the Ironic API identifies the correct Redfish URI to deploy for the RedFish Virtual Media API so that hardware relying on either the deprectaed or newer URI could be provisioned. (link: https://issues.redhat.com/browse/OCPBUGS-30171[*OCPBUGS-30171*]

Reviewer 2: ```suggestion
* Previously, newer versions of Redfish used Manager resources to deprecate the Uniform Resource Identifier (URI) for the RedFish Virtual Media API. This caused any hardware that used the newer Redfish URI for Virtual Media to not be provisioned. With this release, the Ironic API identifies the correct Redfish URI to deploy for the RedFish Virtual Media API so that hardware relying on either the deprectaed or newer URI could be provisioned. (link:https://issues.redhat.com/browse/OCPBUGS-30171[*OCPBUGS-30171*])
```

Author: :heavy_check_mark:

> * Previously, the Bare Metal Operator (BMO) was not using a leader lock to control incoming and outgoing operator pod traffic. After an OpenShift `Deployment` object included a new operator pod, the new pod competed with system resources, such as the `ClusterOperator` status, and this terminated any outgoing operator pods. This issue also impacted clusters that do not include any bare-metal nodes. With this release, the BMO includes a leader lock to manage new pod traffic, and this fix resolves the competing pod issue. (link:https://issues.redhat.com/browse/OCPBUGS-25766[*OCPBUGS-25766*])

Reviewer 2: Operator needs to be capitalized here:
```suggestion
* Previously, the Bare Metal Operator (BMO) was not using a leader lock to control incoming and outgoing operator pod traffic. After an OpenShift `Deployment` object included a new Operator pod, the new pod competed with system resources, such as the `ClusterOperator` status, and this terminated any outgoing Operator pods. This issue also impacted clusters that do not include any bare-metal nodes. With this release, the BMO includes a leader lock to manage new pod traffic, and this fix resolves the competing pod issue. (link:https://issues.redhat.com/browse/OCPBUGS-25766[*OCPBUGS-25766*])
```

Author: :heavy_check_mark:
